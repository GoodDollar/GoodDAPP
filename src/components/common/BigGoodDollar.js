// @flow
import React from 'react'
import BigNumber from './BigNumber'
import { weiToMask } from '../../lib/wallet/utils'

export default ({ number, ...props }: { number: any, props: any }) => (
  <BigNumber number={weiToMask(number)} unit={'GD'} {...props} />
)
