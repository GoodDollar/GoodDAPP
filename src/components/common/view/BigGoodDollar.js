// @flow
import React from 'react'
import { weiToMask } from '../../../lib/wallet/utils'
import Section from '../layout/Section'
import BigNumber from './BigNumber'
import Text from './Text'

type Props = { number: any, props?: {} }

const GOOD_SIGN_SIZE = 18
const DIFF_FACTOR = 0.5
const DOLLAR_SIGN_SIZE = GOOD_SIGN_SIZE - DIFF_FACTOR

/**
 * Receives wei and shows as G$ using BigNumber component
 * @param {Props} params
 * @param {Number} params.number
 * @returns {React.Node}
 */
const BigGoodDollar = ({ number, formatter, testID, ...props }: Props) => {
  const numberFormatter = formatter || weiToMask
  number = number === undefined ? '-.--' : numberFormatter(number)

  return (
    <BigNumber number={number} {...props} testID={testID}>
      {props.unit ? null : <GDUnits {...props} />}
    </BigNumber>
  )
}

const GDUnits = props => {
  const { fontSize, ...bigNumberUnitProps } = props.bigNumberUnitProps || {}
  const goodSignFontSize = fontSize || GOOD_SIGN_SIZE
  const dollarSignFontSize = fontSize ? fontSize - DIFF_FACTOR : DOLLAR_SIGN_SIZE

  return (
    <Section.Row style={props.bigNumberUnitStyles}>
      <Text
        color={props.color || 'gray'}
        fontSize={goodSignFontSize}
        fontFamily="slab"
        fontWeight="bold"
        textAlign="right"
        lineHeight={props.lineHeight || 22}
        {...bigNumberUnitProps}
      >
        G
      </Text>
      <Text
        color={props.color || 'gray'}
        fontSize={dollarSignFontSize}
        fontFamily="slab"
        fontWeight="bold"
        textAlign="right"
        lineHeight={props.lineHeight || 22}
        {...bigNumberUnitProps}
      >
        $
      </Text>
    </Section.Row>
  )
}

export default BigGoodDollar
