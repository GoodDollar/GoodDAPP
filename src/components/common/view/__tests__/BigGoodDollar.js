import React from 'react'
import renderer from 'react-test-renderer'
import BigGoodDollar from '../BigGoodDollar'
import { withThemeProvider } from '../../../../__tests__/__util__'

// Note: test renderer must be required after react-native.

describe('BigGoodDollar', () => {
  const WrappedBigGoodDollar = withThemeProvider(BigGoodDollar)

  it('matches snapshot', async () => {
    let component

    // eslint-disable-next-line require-await
    await renderer.act(async () => (component = renderer.create(<WrappedBigGoodDollar number={1005} />)))
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
