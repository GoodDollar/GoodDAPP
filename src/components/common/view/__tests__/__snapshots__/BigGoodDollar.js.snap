// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigGoodDollar matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "baseline",
        "WebkitBoxAlign": "baseline",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "horizontal",
        "WebkitFlexDirection": "row",
        "alignItems": "baseline",
        "display": "flex",
        "flexDirection": "row",
        "msFlexAlign": "baseline",
        "msFlexDirection": "row",
      }
    }
  >
    <div
      className="css-text-901oao"
      dir="auto"
      style={
        Object {
          "color": "rgba(85,85,85,1.00)",
          "direction": "ltr",
          "fontFamily": "Roboto Slab",
          "fontSize": "36px",
          "fontWeight": "700",
          "lineHeight": "30px",
          "marginRight": "2px",
          "textAlign": "right",
          "textDecoration": "none",
          "textTransform": "none",
        }
      }
    >
      10.05
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitBoxPack": "justify",
          "WebkitFlexDirection": "row",
          "WebkitJustifyContent": "space-between",
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "msFlexAlign": "center",
          "msFlexDirection": "row",
          "msFlexPack": "justify",
        }
      }
    >
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(85,85,85,1.00)",
            "direction": "ltr",
            "fontFamily": "Roboto Slab",
            "fontSize": "18px",
            "fontWeight": "700",
            "lineHeight": "22px",
            "textAlign": "right",
            "textDecoration": "none",
            "textTransform": "none",
          }
        }
      >
        G
      </div>
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(85,85,85,1.00)",
            "direction": "ltr",
            "fontFamily": "Roboto Slab",
            "fontSize": "18px",
            "fontWeight": "700",
            "lineHeight": "22px",
            "textAlign": "right",
            "textDecoration": "none",
            "textTransform": "none",
          }
        }
      >
        $
      </div>
    </div>
  </div>
</div>
`;
